src/TrezorKeyring.ts

```diff
@@ -290,28 +347,35 @@ class TrezorKeyring extends EventEmitter {
    * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  async _signTransaction(address, chainId, tx, handleSigning) {
-    let transaction;
+  private async _signTransaction(
+    address: string,
+    chainId: number,
+    tx: TypedTransaction | OldEthJsTransaction,
```

Comment:
Although it will be a bit verbose, this looks like a good spot for a function overload

Reply:
@andrewpeters9 what do you think about a solution with type generics?

```
async #signTransaction<T extends TypedTransaction | OldEthJsTransaction>(
    address: string,
    chainId: number,
    tx: T,
    handleSigning: (tx: EthereumSignedTx) => T,
): Promise<T>
```